lametro/models.py

```diff
@@ -263,41 +264,95 @@ def upcoming_board_meeting(cls):
+        scheduled to start in the last six hours, or in the next five minutes.
+
+        Fun fact: The longest Metro meeting on record is 5.38 hours long (see
+        issue #251). Hence, we check for meetings scheduled to begin up to six
+        hours ago.
+
+        Used to determine whether to check Granicus for streaming meetings.
         '''
-        Discover and return events in progress.
+        six_hours_ago = cls._time_ago(hours=6)
+        five_minutes_from_now = cls._time_from_now(minutes=5)
 
-        The maximum recorded meeting duration is 5.38 hours, according to the
-        spreadsheet provided by Metro in issue #251. So, to determine initial
-        list of possible current events, we look for all events scheduled
-        in the past 6 hours.
+        return cls.objects.filter(start_time__gte=six_hours_ago,
+                                  start_time__lte=five_minutes_from_now)\
+                           .exclude(status='cancelled')
 
-        A meeting displays as "current" if:
 
-        (1) it started in the last six hours, or it starts in the next five
-            minutes (determined by this method); and
-        (2) it started less than 55 minutes ago, and the previous meeting has
-            ended (determined by `calculate_current_meetings`); or
-        (3) Legistar indicates it is in progress (detemined by `calculate_
-            current_meetings`).
+    @classmethod
+    def _streaming_meeting(cls):
+        '''
+        Granicus provides a running events endpoint that returns an array of
+        GUIDs for streaming meetings. Metro events occur one at a time, but two
+        GUIDs appear when an event is live: one for the English audio, and one
+        for the Spanish audio.
+
+        Hit the endpoint, and return the corresponding meeting, or an empty
+        queryset.
+        '''
+        running_events = requests.get('http://metro.granicus.com/running_events.php')
+
+        for guid in running_events.json():
+            # We get back two GUIDs, but we won't know which is the English
+            # audio GUID stored in the 'guid' field of the extras dict. Thus,
+            # we iterate.
+            meeting = cls.objects.filter(extras__guid=guid.upper())
+
+            if meeting:
+                return meeting
 
-        This method returns a list (with zero or more elements).
+        return cls.objects.none()
 
-        To hardcode current event(s) for testing, use these examples:
-        return LAMetroEvent.objects.filter(start_time='2017-06-15 13:30:00-05')
-        return LAMetroEvent.objects.filter(start_time='2017-11-30 11:00:00-06')
+
+    @classmethod
+    def current_meeting(cls):
         '''
-        from .utils import calculate_current_meetings  # Avoid circular import
+        If there is a meeting scheduled to begin in the last six hours or in
+        the next five minutes, hit the running events endpoint. If there is a
```

Comment:
This is a pretty good explanation of what the code is doing, but it would be even better if it explained why.

In particular, if there is not streaming_meeting, then why do want to show the meetings that started within the last 20 minutes.

Is the idea that streaming_meetings is not completely reliable to indicate that a a meeting is happening right now?

Reply:
no, per metro, events "regularly" start (e.g., show up as streaming) later than their scheduled start time. if we are only looking for meetings five minutes in the future, a meeting that does not start on time will not appear as current. 

so, if there is not a streaming meeting, we also look for meetings scheduled to start up to 20 minutes ago (i consulted with omar on this buffer), so meetings that start late are still returned as current in the interim.

Reply:
@fgregg i expounded on this in the docstring

Reply:
Okay, that's the kind of thing that would be good to have in your comment. This code would be clearer to me if you used if elif else constructs instead of returns.

```python
     if scheduled_meetings:
            streaming_meeting = cls._streaming_meeting()

            if streaming_meeting:
                current_meetings = streaming_meeting
            else:
                # explanation that sometimes a meeting stream shows up late
                twenty_minutes_ago = cls._time_ago(minutes=20)

                # '.annotate' adds a boolean field, 'is_board_meeting'. We want to
                # show board meetings first, so order in reverse, since False (0)
                # comes before True (1).
                current_meetings = scheduled_meetings.filter(start_time__gte=twenty_minutes_ago)\
                                     .annotate(is_board_meeting=RawSQL("name like %s", ('%Board Meeting%',)))\
                                     .order_by('-is_board_meeting')
      else:
           current_meetings = cls.objects.none()

      return current_meetings
```